UG mass reformatting: have addressed some issues (#320)
AY2425S1-CS2103-F09-1/tp: docs/UserGuide.md

* Items with `…`​ after them can be used from one to multiple times.<br>
  e.g. `r/ROLE…​` can be used twice as `r/External r/Marketing` etc.

* Parameters can be in any order.<br>
  e.g. if the command specifies `n/NAME e/EMAIL`, `e/EMAIL n/NAME` is also acceptable.

* Extraneous parameters **//& OR FIELDS? BUT SOUND WEIRD** for commands that do not take in parameters (such as `exit` and `clear`) will be ignored.<br>
  e.g. if the command specifies `exit 123`, it will be interpreted as `exit`.<br>
//& ARE HELP AND LIST CONSIDERED "COMMANDS THAT DO NOT TAKE IN PARAMETERS?"
// no

* The COMMAND_WORD is **case-insensitive**. (eg. add; ADD; aDd are all interpreted as ADD FEATURE)

* A limit of 1000 characters //& REPHRASE IF NEEDED

* If you are using a PDF version of this document, be careful when copying and pasting commands that span multiple lines as space characters surrounding line-breaks may be omitted when copied over to the application.
  </box>

<div style="page-break-after: always;"></div>

### Adding a contact: `add`
Format:<br>`add n/NAME th/TELEGRAM_HANDLE e/EMAIL s/STUDENT_STATUS r/ROLE…​ [nn/NICKNAME]`

<box type="definition" seamless>

> **Action:** Adds a contact to the address book. (parameters do not need to be in order, NICKNAME is optional)

Reviewer: ```suggestion
**Action:** Adds a contact to the address book. (parameters do not need to be in order, nn/NICKNAME is optional)
```

</box>

<box type="tip" seamless>

> **Tip:** A contact must have at least one role which matches\* the following:

Reviewer: This suggests that only one of the roles need to match the role list

Reviewer 2: **Tip:** A contact must have at least one role. Roles should match\* the following:

> **Tip:** A contact must have at least one role which matches\* the following:

Reviewer 2: ```suggestion
**Tip:** A contact must have at least one role. Roles should match\* the following:
```

* `add n/Alex Yeoh th/alexyeoh123 e/alexyeoh@example.com ss/masters r/President r/Admin`
![result for 'add'](images/addAlexYeohResult.png)

Note:

> * `add` command will fail if you try to add a person which already exists. You may refer to [invalid contacts](#what-is-considered-as-invalid-contacts) to see what counts as same person.

Reviewer: ```suggestion
* `add` command will fail if you try to add a person who is already in the address book. You may refer to [invalid contacts](#what-is-considered-as-invalid-contacts) to see what is considered as having the same identity.
```

> * `add` command will fail if you attempt to add person with a duplicate field as an existing contact, please refer to [invalid contacts](#what-is-considered-as-invalid-contacts) for more information.

Reviewer: ```suggestion
* `add` command will fail if you attempt to add a person with a duplicate field as an existing contact, please refer to [invalid contacts](#what-is-considered-as-invalid-contacts) for more information.
```

<div style="page-break-after: always;"></div>

### Editing a contact: `edit`

<box type="definition" seamless>

**Action:** Edits an existing contact in the address book. <br> You may edit by specifying the index of 

> the contact of the displayed address book that you wish to edit, or you may edit by specifying the full name

Reviewer: ```suggestion
the contact in the displayed address book, or by specifying the full name
```

> of the contact that is currently displayed that you wish to edit. <br>

Reviewer: ```suggestion
of any currently displayed contact that you wish to edit. <br>
```

</box>

In both methods of editing:

> * At least one of the optional fields must be provided.

Reviewer: ```suggestion
* Only contacts shown in the displayed list can be edited.
* At least one of the optional fields must be provided.
```

* Existing values will be updated to the input values.
* When editing roles, the existing roles of the contact will be removed.

#### Edit by index
Format:<br> `edit INDEX [n/NAME2] [th/TELEGRAM_HANDLE] [e/EMAIL] [ss/STUDENT_STATUS] [r/ROLE]…​ [nn/NICKNAME]`

* Edits the contact at the specified `INDEX`. 
* The index refers to the index number shown in the displayed contact list. 
* The index **must be a positive integer** 1, 2, 3, …​

Examples:<br>

*  `edit 1 th/johndoe123 e/johndoe@example.com`<br>Edits the telegram handle and email address of the 1st displayed contact to be `johndoe123` and `johndoe@example.com` respectively.<br>

> *  `edit 2 r/Admin r/President` <br> Edits the roles of the 2nd displayed contact to be `Admin` and `President`, this removes all existing roles user has.

Reviewer: ```suggestion
*  `edit 2 r/Admin r/President` <br> Edits the roles of the 2nd displayed contact to be `Admin` and `President`, this replaces all existing roles of the contact.
```

#### Edit by name
Format:<br> `edit NAME [n/NAME2] [th/TELEGRAM_HANDLE] [e/EMAIL] [ss/STUDENT_STATUS] [r/ROLE]…​ [nn/NICKNAME]`
* Edits the contact specified by `NAME`.
* The `NAME` has to be an exact match to an existing contact which is displayed.
* The `NAME` is not case-sensitive.

Examples:
*  `edit alex yeoh th/johndoe123 e/johndoe@example.com` <br> Edits the telegram handle and email address of the contact `Alex Yeoh` to be `johndoe123` and `johndoe@example.com` respectively.
![result for 'edit FULL_NAME'](images/editAlexYeohResult.png)

<box type="tip" seamless>

Note:
* `NAME` refers to the full name of contact displayed (to edit) while `n/NAME2` is an input to change the contact's name to (to change to).

> * `edit` command will fail if you enter a duplicate field that an existing contact has as seen in section [invalid contacts](#what-is-considered-as-invalid-contacts)

Reviewer: ```suggestion
* `edit` command will fail if you enter a duplicate field as an existing contact as seen in section [invalid contacts](#what-is-considered-as-invalid-contacts)
```

* In the event of `NAME` matching to multiple contacts that exist, you will have to `find NAME` followed by `edit INDEX PARAMETER...` to choose which contact of the specified `NAME` you wish to edit.
</box>

<div style="page-break-after: always;"></div>

### Deleting a contact: `delete`
<box type="definition" seamless>

**Action:**
Deletes the specified contact from the address book.<br> You may delete by specifying the index of

> the contact of the displayed address book that you wish to delete, or you may delete by specifying the full name

Reviewer: ```suggestion
the contact in the displayed address book, or you by specifying the full name
```

> of the contact that is currently displayed that you wish to delete. <br>

Reviewer: ```suggestion
of any currently displayed contact that you wish to delete. <br>
```

Format `delete NAME` or `delete n/NAME`

* Delete the contact whose name matches the specified `NAME`.

> * The `NAME` has to be an exact match to an existing contact which is displayed.

Reviewer: ```suggestion
* The `NAME` has to be an exact match to an existing contact in the displayed list.
```

</box>

<div style="page-break-after: always;"></div>

### Finding contacts by contact details: `find`

//& ISSUE 261

Format: `find [n/NAME] [th/TELEGRAM_HANDLE] [e/EMAIL] [ss/STUDENT_STATUS] [r/ROLE]…​ [nn/NICKNAME]`
<box type="definition" seamless>

**Action:** Finds contacts whose details matches all given fields.
* Each PREFIX can only be used once (except Role)

> * All parameters are not case-sensitive. e.g. `find r/pResiDent` &rarr; valid

Reviewer: ```suggestion
* All fields are not case-sensitive. e.g. `find r/pResiDent` &rarr; valid
```

Pagination UI:
![ui of pagination](images/pagination_ui.png)

<div style="page-break-after: always;"></div>

### Showing total number of contacts -- Footer status bar

A status bar that contains information about the number of contacts in the list and the address book data file path is attached as a footer
at the bottom of the app. It shows the total number of contacts being listed in the app.

Examples:

> * `find n/Adam` may show only 1 contact out of 19. 1 refers to the found contact, 19 refers to the total contacts in address book.

Reviewer: ```suggestion
* `find n/Adam` may show only 1 contact out of 19. 1 refers to the number of found contacts, 19 refers to the total number of contacts in address book.
```
